module/zfs/vdev.c

```diff
@@ -1347,6 +1347,10 @@ vdev_open(vdev_t *vd)
 		vd->vdev_max_asize = max_asize;
 		if (vd->vdev_ashift == 0)
 			vd->vdev_ashift = ashift;
+		else
+			if (vd->vdev_ashift < ASHIFT_MIN ||
+			    vd->vdev_ashift > ASHIFT_MAX)
```

Comment:
At this point, if the ashift is invalid, does that mean the vdev can't be opened? If that is the case, should the vdev have its state set to `VDEV_STATE_CANT_OPEN` with something like `VDEV_AUX_BAD_ASHIFT`?